pydrive2/fs/spec.py

```diff
@@ -542,6 +542,43 @@ def cp_file(self, lpath, rpath, **kwargs):
             buffer = io.BytesIO(stream.read())
             self.upload_fobj(buffer, rpath)
 
+    @_gdrive_retry
+    def mv(self, path1, path2, recursive=False, maxdepth=None, **kwargs):
+
+        if maxdepth:
+            raise NotImplementedError("Max depth move is not supported")
+
+        if self.isdir(path1) and not recursive:
```

Comment:
Q: how do we handle recursive after all? Why do we need this extra check. Is it only as a safeguard to check the intention? I'm asking since even the `isdir` check can be expensive (extra call).

Reply:
If you move a folder, the content is moved with it. So is kinda always recursive.

The check is there to avoid that a user wanted to move a file (`recursive=False`) but by accident the source is a directory.

We could also remove `recursive` and always assume that the user want to move the entire folder. This way, there is no need to check if the source is a directory.

Reply:
I would say yes (to avoid that extra API call). Also data does not disappear anywhere, it's not that dangerous.

how does the local filesystem fsspec implementation approach this? 

unix `mv` doesn't require the recursive flag afair

Reply:
I did that to not modify the default method signature of `ffspec`:
https://github.com/fsspec/filesystem_spec/blob/2633445fc54797c79a7ac96d213bd24dfbdfcdc2/fsspec/spec.py#L917-L920
In this case, the flags are passed down to copy.


But yes, if it's allowed to change the signature, we can also remove `recursive` and the `isDir` check.

Reply:
I would just ignore it. We are not allowed to change the default.

We can also check still how they implement the local fs. Is it unix way or it check the recursive flag.

Reply:
> We can also check still how they implement the local fs.  Is it unix way or it check the recursive flag.

Where?